brave/brave-core
# feat(wallet): Balance Details Modal

Add banner when user has a pending BTC balance, tapping banner will show modal displaying `Available`, `Pending`, and `Total` balance for BTC.

Resolves https://github.com/brave/brave-browser/issues/37640

## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

1. Create Bitcoin Testnet account
2. Fund one of your account from a Bitcoin Testnet Faucet, or if https://github.com/brave/brave-browser/issues/37958 has landed you can send BTC in from of your accounts to another.
3. Go the Accounts tab and view details for that account
4. You should see an Banner below BTC asset row stating that Some balances may be unavailable
5. Tap on banner / View Details, you should then see the Balance Details Modal
6. Go to the Portfolio tab
7. You should see an Banner below BTC asset row stating that Some balances may be unavailable

![Portfolio - no grouping](https://github.com/brave/brave-core/assets/5314553/df495806-dc7b-4a3f-b7b2-f44e68f1b67a) | ![Portfolio - account grouping](https://github.com/brave/brave-core/assets/5314553/db2bce1c-7dcd-4ddb-b7e4-3760bf473944) | ![Portfolio - network grouping](https://github.com/brave/brave-core/assets/5314553/df139b65-8049-4faf-897f-2ee568224102) | ![Portfolio - account grouping - btc balance details](https://github.com/brave/brave-core/assets/5314553/493a1723-ba48-4daf-871b-419ce46f887a)
--|--|--|--
![Account details](https://github.com/brave/brave-core/assets/5314553/c8a93f2b-35c5-46a8-94eb-94e7bf4f6f2b) | ![Account details - btc balance details](https://github.com/brave/brave-core/assets/5314553/1bfc14f2-4aa0-4240-b539-0e04c600baeb)
